io/native/src/main/scala/fs2/io/net/tls/S2nConnection.scala

```diff
@@ -114,7 +124,8 @@ private[tls] object S2nConnection {
         }.productL {
           val reads = F.delay(readTasks.get).flatten
           val writes = F.delay(writeTasks.get).flatten
-          reads.both(writes)
+          val pkeyOps = F.delay(privateKeyTasks.get).flatten
+          reads.both(writes).both(pkeyOps)
```

Comment:
As-written there's a potential for memory-leak here 🤔 `s2n_negotiate` allocates memory for private-key operations we are expected to cleanup. If we are cancelled before `pkeyOps` begins executing that cleanup will be skipped.